File: Scripts/inspect_sqlite.js
#!/usr/bin/env node
import sqlite3 from 'sqlite3'
import path from 'path'
import fs from 'fs'

function inspect(dbPath, limit = 5) {
  if (!fs.existsSync(dbPath)) {
    console.error(`ERROR: file not found: ${dbPath}`)
    process.exit(1)
  }

  const sqlite = sqlite3.verbose()
  const db = new sqlite.Database(dbPath, sqlite3.OPEN_READONLY, (err) => {
    if (err) {
      console.error('ERROR opening DB:', err.message)
      process.exit(1)
    }
  })

  db.serialize(() => {
    console.log(`Inspecting: ${dbPath}\n`)
    db.all("SELECT name, sql FROM sqlite_master WHERE type='table' AND name NOT LIKE 'sqlite_%' ORDER BY name", (err, tables) => {
      if (err) {
        console.error('ERROR reading sqlite_master:', err.message)
        db.close()
        process.exit(1)
      }

      if (!tables || tables.length === 0) {
        console.log('No user tables found.')
        db.close()
        return
      }

      (async () => {
        for (const t of tables) {
          console.log(`Table: ${t.name}`)
          console.log('Schema:')
          console.log(t.sql)
          await new Promise((resolve) => {
            db.get(`SELECT COUNT(*) as c FROM '${t.name}'`, (err2, rowCount) => {
              if (err2) {
                console.log('  Count: ERROR', err2.message)
                resolve()
                return
              }
              console.log('  Rows:', rowCount.c)
              db.all(`SELECT * FROM '${t.name}' LIMIT ?`, [limit], (err3, rows) => {
                if (err3) {
                  console.log('  Sample rows: ERROR', err3.message)
                } else {
                  console.log('  Sample rows:')
                  if (!rows || rows.length === 0) console.log('    (no rows)')
                  else rows.forEach(r => console.log('   ', JSON.stringify(r, null, 2)))
                }
                console.log('\n---\n')
                resolve()
              })
            })
          })
        }
        db.close()
      })()
    })
  })
}

if (process.argv.length < 3) {
  console.error('Usage: node inspect_sqlite.js <path-to-db> [limit]')
  process.exit(2)
}

const dbPath = process.argv[2]
const limit = process.argv[3] ? parseInt(process.argv[3], 10) : 5
inspect(dbPath, limit)

File: Scripts/inspect_sqlite.py
#!/usr/bin/env python3
import sqlite3
import sys
import json
import os

def inspect(db_path, limit=5):
    if not os.path.exists(db_path):
        print(f"ERROR: file not found: {db_path}")
        return 1

    conn = sqlite3.connect(db_path)
    conn.row_factory = sqlite3.Row
    cur = conn.cursor()

    print(f"Inspecting: {db_path}\n")

    cur.execute("SELECT name FROM sqlite_master WHERE type='table' AND name NOT LIKE 'sqlite_%' ORDER BY name")
    tables = [r[0] for r in cur.fetchall()]
    if not tables:
        print("No user tables found.")
        return 0

    for table in tables:
        try:
            cur.execute(f"SELECT COUNT(*) as c FROM '{table}'")
            count = cur.fetchone()[0]
        except Exception as e:
            count = f"ERROR: {e}"

        print(f"Table: {table} (rows: {count})")

        try:
            cur.execute(f"SELECT * FROM '{table}' LIMIT ?", (limit,))
            rows = [dict(r) for r in cur.fetchall()]
        except Exception as e:
            rows = f"ERROR: {e}"

        print("Sample rows:")
        if isinstance(rows, str):
            print(rows)
        else:
            if len(rows) == 0:
                print("  (no rows)")
            else:
                for r in rows:
                    print(json.dumps(r, ensure_ascii=False, indent=2))

        print("\n---\n")

    conn.close()
    return 0

def main():
    if len(sys.argv) < 2:
        print("Usage: inspect_sqlite.py <path-to-db> [limit]")
        return 2
    db_path = sys.argv[1]
    limit = int(sys.argv[2]) if len(sys.argv) >= 3 else 5
    return inspect(db_path, limit)

if __name__ == '__main__':
    raise SystemExit(main())
